flow/layers/layer.h

```diff
@@ -66,6 +66,11 @@ struct PrerollContext {
   float total_elevation = 0.0f;
   bool has_platform_view = false;
   bool is_opaque = true;
+#if defined(OS_FUCHSIA)
+  // True if, during the traversal so far, we have seen a child_scene_layer.
+  // Informs whether a layer needs to be system composited.
+  bool child_scene_layer_exists_below = false;
```

Comment:
Mike: You can (and should) fix this bug as part of this CL by modifying the logic in ContainerLayer::Preroll to reset the needs_system_composite flag similar to the way it handles the has_platform_view flag

Reply:
I've been thinking about this more in terms of traversal order. Although all the layers are stored in a tree, there is (I assume) a deterministic traversal order, which defines the paint order.

Based on that, if child_scene_layer_exists_below is true, all physicalshapelayers painted afterward will be system composited.

that said, should performance overlay also be system composited if there is embedded content below?

Reply:
Ah, I misread the bug a bit.  I agree, the proper fix is to system_composite the performance overlay if there is embedded content below.

Reply:
Discussed offline why we need to do things differently than has_platform_view.

Going to hold off any changes to performance overlay in this change since there's no regression (i.e. it's not popped up to a separate layer on ToT either).